Sources/SwiftKafka/KafkaClient.swift

```diff
@@ -0,0 +1,68 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the swift-kafka-gsoc open source project
+//
+// Copyright (c) 2022 Apple Inc. and the swift-kafka-gsoc project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of swift-kafka-gsoc project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Crdkafka
+import Logging
+
+/// Base class for ``KafkaProducer`` and ``KafkaConsumer``, which is used to handle the connection to the Kafka ecosystem.
+public class KafkaClient {
+    // Default size for Strings returned from C API
+    static let stringSize = 1024
+
+    private var _kafkaHandle: OpaquePointer?
+    private var _clientType: rd_kafka_type_t
+    private var _config: KafkaConfig
+
+    static let logger = Logger(label: "SwiftKafkaLogger")
```

Comment:
We should get the logger passed in and not instantiate one of our own. Make sure to give this guide a read: https://www.swift.org/server/guides/libraries/log-levels.html